contrib/hack/update-codecgen.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
```

Comment:
minor thing, but I think this file should be under `build/` since stuff under `contrib/` isn't "essential" and this clearly is - and we need it for building.